### PR TITLE
Adding staircase entangler layer

### PR DESCRIPTION
**Pull request by bpenchas**

### CODE REVIEW

File: tensornetwork/tn_keras/entangler.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    def is_perfect_root(n, n_nodes):
+      root = n**(1. / n_nodes)
+      return round(root)**n_nodes == n
+
+    super(DenseEntangler, self).build(input_shape)
+
+    # Ensure input dim and output dim match
+    assert input_shape[-1] == self.output_dim, \
+      f'Input dim {input_shape[-1]} and output dim \
+      {self.output_dim} must match.'
```
Comment (Thenerdstation) [inline tensornetwork/tn_keras/entangler.py:100]: Don't use `\` to break lines. Prefer to wrap code in `()` instead.

Comment (Thenerdstation) [reply tensornetwork/tn_keras/entangler.py:100]: Surprised the linter didn't complain...

File: tensornetwork/tn_keras/entangler.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+               use_bias: Optional[bool] = True,
+               activation: Optional[Text] = None,
+               kernel_initializer: Optional[Text] = 'glorot_uniform',
+               bias_initializer: Optional[Text] = 'zeros',
+               **kwargs) -> None:
+
+    if 'input_shape' not in kwargs and 'input_dim' in kwargs:
+      kwargs['input_shape'] = (kwargs.pop('input_dim'),)
+
+    assert num_legs > 2, 'Need at least 2 legs to create Entangler'
```
Comment (markomernick) [inline tensornetwork/tn_keras/entangler.py:70]: Can you add a printout of what the received value was here?

Comment (bpenchas) [reply tensornetwork/tn_keras/entangler.py]: Updated!

**Pull request merged**